File: java_src/DH3DLibrary/src/jp/spec/darkhorse2/dh3d/base/Skin.java
/**
 * 
 */
package jp.spec.darkhorse2.dh3d.base;

/**
 * @author Norihiro
 *
 */
public class Skin {
	public DHVector3 position;
	public DHVector3 normal;
	public TextureUV textureUV;
	public DHVector3 currentPosition;
	public DHVector3 currentNormal;
	public TextureUV currentTextureUV;
	public RenderGroup renderGroup;
	public int[] boneNum;
	public Bone[] bones;
	public float[] skinWeight;
	public int[] boneIndex;
	public int edge;
	public DHMatrix tempMatrix;
	
	public Skin(){
		
	}
}

File: js/base/Skin.js
/*--------------------------------------------------------------------------------
 * DH3DLibrary Skin.js v0.2.0
 * Copyright (c) 2010-2012 DarkHorse
 *
 * DH3DLibrary is freely distributable under the terms of an MIT-style license.
 * For details, see the DH3DLibrary web site: http://darkhorse2.0spec.jp/dh3d/
 *
 *------------------------------------------------------------------------------*/
var Skin = Class.create({
  // Vertex
  position: null,          // vec3
  normal: null,            // vec3
  textureUV: null,         // vec2

  currentPosition: null,
  currentNormal: null,
  currentTextureUV: null,

  renderGroup: null,
  boneNum: null,
  bones: null,
  skinWeight: null,
  boneIndex: null,

  // etc.
  edge: 0,

  tempMatrix: new DHMatrix(),

  initialize: function() {
    this.currentPosition = new DHVector3();
    this.currentNormal = new DHVector3();
    this.currentTextureUV = new TextureUV();

    this.bones = $A();
    this.boneNum = $A();
    this.skinWeight = $A();
    this.boneIndex = $A();
  },
});


